pig/src/main/java/com/netflix/iceberg/pig/IcebergInputFormat.java

```diff
@@ -0,0 +1,260 @@
+  }
+
+  @Override
+  @SuppressWarnings("unchecked")
+  public List<InputSplit> getSplits(JobContext context) throws IOException, InterruptedException {
+    if (splits != null) {
+      LOG.info("Returning cached splits: " + splits.size());
+      return splits;
+    }
+
+    splits = Lists.newArrayList();
+
+    TableScan scan = table.newScan().select(SNAPSHOT_COLUMNS);
+
+    //Apply Filters
+    Expression filterExpression = (Expression) ObjectSerializer.deserialize(context.getConfiguration().get(ICEBERG_FILTER_EXPRESSION));
+
+    if (filterExpression != null) {
+      LOG.info("Filter Expression: " + filterExpression);
+      scan = scan.filter(filterExpression);
+    }
+
+    //Wrap in Splits
+    scan.planTasks().forEach((scanTask) -> splits.add(new IcebergSplit(scanTask)));
+
+    return splits;
+  }
+
+  @Override
+  public RecordReader<Void, T> createRecordReader(InputSplit split, TaskAttemptContext context) throws IOException, InterruptedException {
+    return new IcebergRecordReader<>();
+  }
+
+  public static class IcebergSplit extends InputSplit implements Writable {
+    private CombinedScanTask task;
+
+    public IcebergSplit() {
+
+    }
+
+    public IcebergSplit(CombinedScanTask task) {
+      this.task = task;
+    }
+
+    @Override
+    public long getLength() throws IOException, InterruptedException {
+      return task.files().stream().mapToLong(FileScanTask::length).sum();
+    }
+
+    @Override
+    public String[] getLocations() throws IOException, InterruptedException {
+      return new String[0];
+    }
+
+    @Override
+    public void write(DataOutput out) throws IOException {
+      byte[] data = SerializationUtils.serialize(this.task);
+      out.writeInt(data.length);
+      out.write(data);
+    }
+
+    @Override
+    public void readFields(DataInput in) throws IOException {
+      byte[] data = new byte[in.readInt()];
+      in.readFully(data);
+
+      this.task = (CombinedScanTask) SerializationUtils.deserialize(data);
+    }
+  }
+
+  public class IcebergRecordReader<T> extends RecordReader<Void, T> {
+    private IcebergSplit split;
+    private TaskAttemptContext context;
+    private ReadSupport<T> readSupport;
+
+    private Iterator<FileScanTask> tasks;
+    private FileScanTask currentTask;
+
+    private Iterator<T> recordIterator;
+    private T currentRecord;
+
+    @Override
+    public void initialize(InputSplit split, TaskAttemptContext context) throws IOException, InterruptedException {
+      this.split = (IcebergSplit) split;
+      this.context = context;
+
+      CombinedScanTask ctask = ((IcebergSplit) split).task;
+      tasks = ctask.files().iterator();
+
+      advance();
+    }
+
+    @SuppressWarnings("unchecked")
+    private boolean advance() throws IOException {
+      if (!tasks.hasNext()) {
+        return false;
+      }
+      currentTask = tasks.next();
+
+      Schema tableSchema = (Schema) ObjectSerializer.deserialize(context.getConfiguration().get(ICEBERG_SCHEMA));
```

Comment:
[`SchemaParser`](https://github.com/Netflix/iceberg/blob/master/core/src/main/java/com/netflix/iceberg/SchemaParser.java) provides conversion to and from JSON, which may be friendlier for debugging than using `ObjectSerializer`.

Reply:
Since this gets injected into xml element text, it's possible that you could have conflicts with it inserted directly.  This is generally safer and the common approach used across many pig loader implementations.

Reply:
Sounds good.